com.unity.shadergraph/Editor/Drawing/Views/GraphEditorView.cs

```diff
@@ -560,7 +586,18 @@ public void HandleGraphChanges()
             }
 
             previewManager.RenderPreviews();
-            m_BlackboardProvider.HandleGraphChanges();
+            m_BlackboardProvider.HandleGraphChanges(wasUndoRedoPerformed);
+
+            if (m_updateInspectorNextFrame)
+            {
+                m_updateInspectorNextFrame = false;
+                m_InspectorView.Update();
+            }
+
+            // #TODO: Graph selection of elements is weird with undo-redo, doesn't always properly trigger the undoRedoPerformed flag from graphObject
```

Comment:
I don't like leaving random TODOs. Either fix it now, or explain that it's weird and move on.